### PR TITLE
feat: Support Poise

### LINKED ISSUE #552: Add a PoiseService for serenity
Poise is a framework for serenity, and it is quite easy to implement. I think that we should implement service for the struct, and not the direct framework object.

### PR DESCRIPTION
**Pull request by Anafabula**
fixes #552
[Example](https://github.com/Anafabula/shuttle-examples/tree/feat/poise-example/poise/hello-world)

`cargo shuttle run` works fine, ~~but so far I haven't been able to deploy _anything_ locally with errors like:~~
<strike>
```
ERROR {error="Build error: failed to get `axum` as a dependency of package `hello-world-fd684ca9-901c-44ff-882b-2b54a0972e6c v0.1.0 (/opt/shuttle/shuttle-builds/hello-world-axum-app)`"} shuttle_deployer::deployment::queue: service build encountered an error
```
</strike>
Deploying on a local instance of shuttle works too when using crates.io directly instead of panamax.

### CODE REVIEW

File: service/src/lib.rs [hunk 1]
```diff
@@ -661,5 +662,22 @@ impl Service for serenity::Client {
 #[cfg(feature = "bot-serenity")]
 pub type ShuttleSerenity = Result<serenity::Client, Error>;
 
+#[cfg(feature = "bot-poise")]
+#[async_trait]
+impl<T, E> Service for std::sync::Arc<poise::Framework<T, E>>
```
Comment (chesedo) [inline service/src/lib.rs:667]: This should work without the `Arc` right?

File: service/src/lib.rs [hunk 2]
```diff
@@ -264,6 +264,7 @@ extern crate shuttle_codegen;
 /// | `ShuttlePoem`                         | web-poem     | [poem](https://docs.rs/poem/1.3.35)         | 1.3.35     | [GitHub](https://github.com/shuttle-hq/examples/tree/main/poem/hello-world)           |
 /// | `Result<T, shuttle_service::Error>`   | web-tower    | [tower](https://docs.rs/tower/0.4.12)       | 0.14.12    | [GitHub](https://github.com/shuttle-hq/examples/tree/main/tower/hello-world)          |
 /// | `ShuttleSerenity`                     | bot-serenity | [serenity](https://docs.rs/serenity/0.11.5) | 0.11.5     | [GitHub](https://github.com/shuttle-hq/examples/tree/main/serenity/hello-world)       |
+/// | `ShuttlePoise`                        | bot-poise    | [poise](https://docs.rs/poise/0.5.2)        | 0.5.2      | TODO                                                                                  |
```
Comment (chesedo) [inline service/src/lib.rs:267]: Todo reminder :smile:

**Review by chesedo — COMMENTED** [review]
Oh nice, thanks for this @Anafabula!! Can you also show an [example](https://github.com/shuttle-hq/examples) of how to use poise with shuttle?

**Comment by oddgrd** [comment]
Cool! It would also be nice to have a `cargo shuttle init` implementation for this, if you have time to add that as well, if not we can do it later no problem. :slightly_smiling_face:  

The code for this is here: https://github.com/shuttle-hq/shuttle/blob/main/cargo-shuttle/src/init.rs

File: service/src/lib.rs (hunk 2 above)
Comment (Anafabula) [reply service/src/lib.rs:267]: Thanks, I made a pull request at shuttle-hq/examples#9

**Comment by Anafabula** [comment]
@oddgrd I added poise to `cargo shuttle init`

File: service/src/lib.rs (hunk 1 above)
Comment (chesedo) [reply service/src/lib.rs:667]: Ah, I see. The `Arc` should be all good then :smile:

**Review by chesedo — APPROVED** [review]

**Review by oddgrd — APPROVED** [review]
Looks good to me too, thanks a lot @Anafabula !

**Pull request merged**